### PR TITLE
Logs and metrics delivery via Stack Monitoring documentation

### PR DESCRIPTION
**Pull request by thbkrkr**
Relates to #4629.

### CODE REVIEW

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+****
+endif::[]
+
+[id="{p}-{page_id}"]
+= Stack Monitoring
+
+You can enable link:https://www.elastic.co/guide/en/elasticsearch/reference/current/monitor-elasticsearch-cluster.html[Stack Monitoring]
+on your Elasticsearch and Kibana to collect and ship their metrics and logs.
+
+You just have to reference the separate monitoring Elasticsearch cluster in the `spec.monitoring` section.
```
Comment (pebrc) [inline docs/advanced-topics/stack-monitoring.asciidoc:14]:
```suggestion
You just have to reference a separate monitoring Elasticsearch cluster in the `spec.monitoring` section.
```

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+          namespace: observability
+    logs:
+      elasticsearchRefs:
+        - name: monitoring
+          namespace: observability
+  count: 1
+EOF
+----
+
+The use of `namespace` is optional if the Elasticsearch cluster and the Kibana are running in the same namespace.
```
Comment (pebrc) [inline docs/advanced-topics/stack-monitoring.asciidoc:64]: Should this be a [callout ](https://github.com/elastic/docs#callouts) in the YAML sample?

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+The monitoring cluster must be managed by ECK in the same Kubernetes cluster as the monitored one.
+
+You can send metrics and logs to two different Elasticsearch monitoring clusters.
+
+You can enable Stack Monitoring only on Elasticsearch or only on Kibana. In the latter case, Kibana will not be available on the Stack Monitoring Kibana page (see link:https://www.elastic.co/guide/en/kibana/current/monitoring-data.html#monitoring-data(View monitoring data in Kibana)).
```
Comment (pebrc) [inline docs/advanced-topics/stack-monitoring.asciidoc:70]:
```suggestion
You can also enable Stack Monitoring only on Elasticsearch or only on Kibana. In the latter case, Kibana will not be available on the Stack Monitoring Kibana page (see link:https://www.elastic.co/guide/en/kibana/current/monitoring-data.html#monitoring-data(View monitoring data in Kibana)).
```

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+== How it works
+
+In the background, Metricbeat and Filebeat are deployed as sidecar containers in the same Pod than Elasticsearch and Kibana.
```
Comment (pebrc) [inline docs/advanced-topics/stack-monitoring.asciidoc:74]:
```suggestion
In the background, Metricbeat and Filebeat are deployed as sidecar containers in the same Pod as Elasticsearch and Kibana.
```

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+Metricbeat is used to collect monitoring metrics and Filebeat to monitor the Elasticsearch log files, collect log events. The two beats are configured to ship data directly to the monitoring cluster(s) using dedicated Elastic users managed by ECK.
```
Comment (pebrc) [inline docs/advanced-topics/stack-monitoring.asciidoc:76]:
```suggestion
Metricbeat is used to collect monitoring metrics and Filebeat to monitor the Elasticsearch log files, collect log events. The two Beats are configured to ship data directly to the monitoring cluster(s) using dedicated Elastic users managed by ECK.
```

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+Elasticsearch and Kibana containers are configured to share the
```
Comment (pebrc) [inline docs/advanced-topics/stack-monitoring.asciidoc:78]: Incomplete sentence?

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        - name: monitoring
+          namespace: observability
+  config:
+    # https://www.elastic.co/guide/en/kibana/current/xpack-security-audit-logging.html
+    xpack.security.audit.enabled: true
+----
+
+== Override Beats Pod Template
+
+You can customize the Filebeat and Metricbeat containers through the Pod template. Your config is merged with our defaults.
```
Comment (pebrc) [inline docs/advanced-topics/stack-monitoring.asciidoc:123]:
```suggestion
You can customize the Filebeat and Metricbeat containers through the Pod template. Your config is merged with  the values of the default Pod template ECK uses.
```

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+    logs:
+      elasticsearchRefs:
+        - name: monitoring
+          namespace: observability
+  config:
+    # https://www.elastic.co/guide/en/kibana/current/xpack-security-audit-logging.html
+    xpack.security.audit.enabled: true
+----
+
+== Override Beats Pod Template
```
Comment (pebrc) [inline docs/advanced-topics/stack-monitoring.asciidoc:121]:
```suggestion
== Override the Beats Pod Template
```

File: docs/advanced-topics/stack-monitoring.asciidoc (hunk 1 above)
Comment (shubhaat) [reply docs/advanced-topics/stack-monitoring.asciidoc:14]: Can a user configure the same Elasticsearch cluster to ship logs and metrics? This is a pattern we currently support in ECE, just curious, if the cluster has to be a separate cluster?

File: docs/advanced-topics/stack-monitoring.asciidoc (hunk 2 above)
Comment (thbkrkr) [reply docs/advanced-topics/stack-monitoring.asciidoc]: Good idea.

File: docs/advanced-topics/stack-monitoring.asciidoc (hunk 6 above)
Comment (thbkrkr) [reply docs/advanced-topics/stack-monitoring.asciidoc:79]: Whoooops. I was about to explain the logs volume sharing for Filebeat. But I'm not sure how far to go in explaining how it works. It seems to me that mentioning the sidecars is the most important part and the rest is secondary.

File: docs/advanced-topics/stack-monitoring.asciidoc (hunk 1 above)
Comment (thbkrkr) [reply docs/advanced-topics/stack-monitoring.asciidoc]: A user cannot configure an Elasticsearch cluster to monitor itself, the monitoring cluster has to be a separate cluster.
This is not yet supported due to a circular dependency issue (https://github.com/elastic/cloud-on-k8s/issues/4627).

Comment (thbkrkr) [reply docs/advanced-topics/stack-monitoring.asciidoc]: I'm adding a note on this limitation.

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+        namespace: observability <1>
+    logs:
+      elasticsearchRefs:
+      - name: monitoring
+        namespace: observability <1>
+  count: 1
+----
+
+<1> The use of `namespace` is optional if the monitoring Elasticsearch cluster and the monitored Elasticsearch cluster are running in the same namespace.
+<2> The use of `namespace` is optional if the Elasticsearch cluster and the Kibana are running in the same namespace.
```
Comment (kilfoyle) [inline docs/advanced-topics/stack-monitoring.asciidoc:63]:
```suggestion
<2> The use of `namespace` is optional if the Elasticsearch cluster and the Kibana instance are running in the same namespace.
```

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+CAUTION: You cannot configure an Elasticsearch cluster to monitor itself, the monitoring cluster has to be a separate cluster.
+
+IMPORTANT: The monitoring cluster must be managed by ECK in the same Kubernetes cluster as the monitored one.
+
+You can send metrics and logs to two different Elasticsearch monitoring clusters.
+
+You can also enable Stack Monitoring only on Elasticsearch or only on Kibana. In the latter case, Kibana will not be available on the Stack Monitoring Kibana page (see link:https://www.elastic.co/guide/en/kibana/current/monitoring-data.html#monitoring-data[View monitoring data in Kibana]).
```
Comment (kilfoyle) [inline docs/advanced-topics/stack-monitoring.asciidoc:71]:
```suggestion
You can also enable Stack Monitoring on Elasticsearch only or on Kibana only. In the latter case, Kibana will not be available on the Stack Monitoring Kibana page (see link:https://www.elastic.co/guide/en/kibana/current/monitoring-data.html#monitoring-data[View monitoring data in Kibana]).
```

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+****
+link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{page_id}.html[View this document on the Elastic website]
+****
+endif::[]
+
+[id="{p}-{page_id}"]
+= Stack Monitoring
+
+You can enable link:https://www.elastic.co/guide/en/elasticsearch/reference/current/monitor-elasticsearch-cluster.html[Stack Monitoring]
+on your Elasticsearch and Kibana to collect and ship their metrics and logs.
```
Comment (kilfoyle) [inline docs/advanced-topics/stack-monitoring.asciidoc:12]:
```suggestion
on Elasticsearch and Kibana to collect and ship their metrics and logs to a dedicated monitoring cluster.
```
In the ESS docs we emphasize "the dedicated monitoring cluster" so maybe we could use that here too.

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+
+You just have to reference a separate monitoring Elasticsearch cluster in the `spec.monitoring` section.
```
Comment (kilfoyle) [inline docs/advanced-topics/stack-monitoring.asciidoc:14]:
```suggestion
To enable stack monitoring, simply reference the monitoring Elasticsearch cluster in the spec.monitoring section of the ECK configuration.
```

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+
+In the background, Metricbeat and Filebeat are deployed as sidecar containers in the same Pod as Elasticsearch and Kibana.
+
+Metricbeat is used to collect monitoring metrics and Filebeat to monitor the Elasticsearch log files and collect log events.
+
+The two Beats are configured to ship data directly to the monitoring cluster(s) using HTTPS and dedicated Elastic users managed by ECK.
+
+== Audit logging
+
+Audit logs are collected and ship to the monitoring cluster referenced in the `monitoring.logs` section when audit logging is enabled (disabled by default).
```
Comment (kilfoyle) [inline docs/advanced-topics/stack-monitoring.asciidoc:92]:
```suggestion
Audit logs are collected and shipped to the monitoring cluster referenced in the `monitoring.logs` section when audit logging is enabled (disabled by default).
```

File: docs/advanced-topics/stack-monitoring.asciidoc [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+      - name: monitoring
+        namespace: observability
+  config:
+    # https://www.elastic.co/guide/en/kibana/current/xpack-security-audit-logging.html
+    xpack.security.audit.enabled: true
+----
+
+== Override the Beats Pod Template
+
+You can customize the Filebeat and Metricbeat containers through the Pod template. Your config is merged with the values of the default Pod template ECK uses.
```
Comment (kilfoyle) [inline docs/advanced-topics/stack-monitoring.asciidoc:133]:
```suggestion
You can customize the Filebeat and Metricbeat containers through the Pod template. Your configuration is merged with the values of the default Pod template that ECK uses.
```

**Review by kilfoyle — APPROVED** [review]
LGTM for the docs! 👍 
I added just a few minor suggestions.

File: docs/advanced-topics/stack-monitoring.asciidoc (hunk 11 above)
Comment (thbkrkr) [reply docs/advanced-topics/stack-monitoring.asciidoc:12]: Sounds good to me.

File: docs/advanced-topics/stack-monitoring.asciidoc (hunk 12 above)
Comment (thbkrkr) [reply docs/advanced-topics/stack-monitoring.asciidoc:14]: `of the ECK configuration` is not really correct here. The ECK operator has a [configuration](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-operator-config.html). Below this sentence, the YAML code corresponds to the Elasticsearch cluster specification and the Kibana instance(s) specification.

As we just said ES & KB in "You can enable Stack Monitoring on your Elasticsearch and Kibana to collect and ship their metrics and logs.", can we say:

To enable stack monitoring, simply reference the monitoring Elasticsearch cluster in the `spec.monitoring` section of their specification.

Comment (kilfoyle) [reply docs/advanced-topics/stack-monitoring.asciidoc:14]: Thanks. I wasn't sure about that configuration part. Your suggested text looks perfect.

File: docs/release-notes/highlights-1.7.0.asciidoc [hunk 15]
```diff
@@ -19,7 +19,7 @@ Starting with this release, the `CustomResourceDefinitions` (CRD) and the `Valid
 
 In this release, the Elasticsearch and Kibana resources have been enhanced to let you specify a reference to a monitoring cluster. When specified, sidecar containers are automatically setup by ECK to ship logs and metrics to the referenced Elasticsearch cluster.
 
-*Add a link to documentation*
+<<{p}-stack-monitoring,Stack Monitoring documentation>>
```
Comment (barkbay) [inline docs/release-notes/highlights-1.7.0.asciidoc:22]:
```suggestion
```

**Review by barkbay — APPROVED** [review]
lgtm

**Pull request merged**